src/char/char_mapif.cpp

```diff
@@ -72,7 +72,7 @@ int chmapif_sendallwos(int sfd, unsigned char *buf, unsigned int len){
  * @return : the number of map-serv the packet was sent to (O|1)
  */
 int chmapif_send(int fd, unsigned char *buf, unsigned int len){
-	if (fd >= 0) {
+	if (session_isValid(fd)) {
```

Comment:
You missed two cases above. In chmapif_sendall and chmapif_sendallwos. Please also change them.